streams/src/test/java/org/apache/kafka/streams/integration/KTableKTableForeignKeyJoinIntegrationTest.java

```diff
@@ -27,673 +25,314 @@
 import org.apache.kafka.streams.state.KeyValueStore;
-import org.apache.kafka.streams.state.QueryableStoreTypes;
-import org.apache.kafka.streams.state.ReadOnlyKeyValueStore;
+import org.apache.kafka.streams.state.Stores;
 import org.apache.kafka.test.IntegrationTest;
 import org.apache.kafka.test.TestUtils;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.junit.ClassRule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 
-import java.io.IOException;
-import java.util.Arrays;
 import java.util.HashMap;
-import java.util.HashSet;
-import java.util.LinkedList;
-import java.util.List;
 import java.util.Map;
 import java.util.Properties;
-import java.util.Set;
 import java.util.function.Function;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.assertArrayEquals;
+import static java.util.Collections.emptyMap;
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+import static org.apache.kafka.common.utils.Utils.mkProperties;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
 
 
-@Category({IntegrationTest.class})
+@Category(IntegrationTest.class)
 public class KTableKTableForeignKeyJoinIntegrationTest {
-    private final static int NUM_BROKERS = 1;
 
-    @ClassRule
-    public final static EmbeddedKafkaCluster CLUSTER = new EmbeddedKafkaCluster(NUM_BROKERS);
-    private final static MockTime MOCK_TIME = CLUSTER.time;
-    private final static String LEFT_TABLE = "left_table";
-    private final static String RIGHT_TABLE = "right_table";
-    private final static String OUTPUT = "output-topic";
-    private static Properties streamsConfig;
-    private KafkaStreams streams;
-    private KafkaStreams streamsTwo;
-    private KafkaStreams streamsThree;
-    private static final  Properties CONSUMER_CONFIG = new Properties();
-    private static final Properties LEFT_PROD_CONF = new Properties();
-    private static final Properties RIGHT_PROD_CONF = new Properties();
+    private static final String LEFT_TABLE = "left_table";
+    private static final String RIGHT_TABLE = "right_table";
+    private static final String OUTPUT = "output-topic";
+    private static final Properties STREAMS_CONFIG = mkProperties(mkMap(
+        mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "ktable-ktable-joinOnForeignKey"),
+        mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "asdf:0000"),
+        mkEntry(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"),
+        mkEntry(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getPath()),
+        mkEntry(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, "0"),
+        mkEntry(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, "100"),
+        mkEntry(StreamsConfig.TOPOLOGY_OPTIMIZATION, StreamsConfig.OPTIMIZE)
```

Comment:
Do we need/want this?

Reply:
Not sure... Maybe it helps verify that the optimizer doesn't mess up the FK join?

Reply:
Not sure either -- maybe we should test the optimization layer in a more structured way... Also, should be test with and without? No need to update the PR. Just brainstorming in this comment.

Reply:
Actually, now that the tests are so fast, I'll just add an optimization parameter so we test it both ways.